docs/workflow.rst

```diff
@@ -0,0 +1,231 @@
+        "HST = loadmol2 host.mol2",
+        "GST = loadmol2 guest.mol2",
+        "DM1 = loadmol2 dm1.mol2",
+        "DM2 = loadmol2 dm2.mol2",
+        "DM3 = loadmol2 dm3.mol2",
+        "model = loadpdb aligned_with_dummy.pdb",
+    ]
+    system.build()
+
+
+Defining Restraints
+-------------------
+
+.. figure :: _static/images/restraints.png
+   :figwidth: 550px
+   :align: center
+
+In APR calculations we apply restraints on the host (or protein) and the guest molecules. The restraints can be grouped
+into four categories: (1) *static restraints*, (2) *varying restraints*, (3) *wall restraints* and (4) *positional
+restraints*.
+
+**(1) Static Restraints**
+
+Static restraints do not change during the whole APR process and do not affect the free-energy. We apply static
+restraints on the host (or protein) molecule to define a guest molecule path. The static restraints are composed of
+distance, angle, and torsional (DAT) restraints based on the choice of anchor atoms. For host-guest systems, we need to
+define three anchor atoms ``[H1,H2,H3]`` and combined with three dummy atoms ``[D1,D2,D3]``, we apply a total of six
+static restraints on the host molecule (three for the translation and three for orientation).
+
+To generate static restraints we use the function ``static_DAT_restraints``. As an example, to apply a distance restraint
+on ``D1`` and ``H1`` with a force constant of 5 kcal/mol/:math:`Å^2` we call
+
+.. code :: python
+
+    from paprika.restraints import static_DAT_restraint
+
+    dist_static = static_DAT_restraint(
+        restraint_mask_list = [D1, H1],
+        num_window_list = windows,  # list: [len(attach_lambda), len(pull_windows), len(release_lambda)]
+        ref_structure = structure,  # Structure file (PDB)
+        force_constant = 5.0,
+    )
+
+**(2) Varying Restraints**
+
+As the name suggests, these restraints change during the APR process. During the attach and release phases, the force
+constants of these restraints changes. In the pull phase, `varying restraints` can have their equilibrium targets change,
+and this can be used as the restraint to pull the guest molecule out of the host molecule. To generate `varying restraints`,
+we use the ``DAT_restraint`` class. The code below shows a restraints `r` that starts from 6.0 Å to 24 Å in the *pull*
+phase and stays restrained at 24 Å during the *release* phase.
+
+.. code :: python
+
+    from paprika.restraints import DAT_restraint
+
+    r = DAT_restraint()
+    r.mask1 = D1
+    r.mask2 = G1
+    r.topology = structure
+    r.auto_apr = True
+    r.continuous_apr = True
+
+    r.attach["target"] = 6.0
+    r.attach["fraction_list"] = attach_lambda
+    r.attach["fc_final"] = 5.0
+
+    r.pull["target_final"] = 24.0
+    r.pull["num_windows"] = len(pull_windows)
+
+    r.release["target"] = 24.0
+    r.release["fraction_list"] = [1.0] * len(release_lambda)
+    r.release["fc_final"] = 5.0
+
+    r.initialize()
+
+
+**(3) Wall Restraints**
```

Comment:
Worth noting this is optional.

Reply:
True. added "(optional)" to the heading.